plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -115,7 +115,7 @@
     private final LoadingCache<HiveTableName, Optional<Table>> tableCache;
     private final LoadingCache<String, List<String>> tableNamesCache;
     private final LoadingCache<TablesWithParameterCacheKey, List<String>> tablesWithParameterCache;
-    private final LoadingCache<HiveTableName, PartitionStatistics> tableStatisticsCache;
+    private final LoadingCache<HiveTableName, AtomicReference<PartitionStatistics>> tableStatisticsCache;
```

Comment:
tableStatisticsCache is not used as a LoadingCache, it?
you can change the field definition to `Cache<...>`